### PR TITLE
Compute engine dedicated logs

### PR DESCRIPTION
**Pull request by simonbrandhof**
Each task of CE has its own log file. This file is stored in the repository data/ 
in order to be kept after server restarts and upgrades. It is accessible 
through the WS api/ce/logs.

Related JIRA ticket is https://jira.sonarsource.com/browse/SONAR-6799

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/log/LogFileRef.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    LogFileRef that = (LogFileRef) o;
+
+    if (!taskUuid.equals(that.taskUuid)) {
+      return false;
+    }
+    return !(componentUuid != null ? !componentUuid.equals(that.componentUuid) : that.componentUuid != null);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/log/LogFileRef.java]: MAJOR Add parentheses to make the operator precedence explicit. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS864)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by sns-seb** [comment]
back to you @simonbrandhof , but no LGTM, several design comments need to be addressed

File: server/sonar-server/src/main/java/org/sonar/server/computation/log/LogFileRef.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      return false;
+    }
+    LogFileRef that = (LogFileRef) o;
+    if (!taskType.equals(that.taskType)) {
+      return false;
+    }
+    if (!taskUuid.equals(that.taskUuid)) {
+      return false;
+    }
+    return !(componentUuid != null ? !componentUuid.equals(that.componentUuid) : that.componentUuid != null);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/log/LogFileRef.java]: MAJOR Add parentheses to make the operator precedence explicit. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS864)

File: server/sonar-server/src/main/java/org/sonar/server/computation/log/LogFileRef.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+   */
+  public String getRelativePath() {
+    if (componentUuid == null) {
+      return format("%s/%s.log", taskType, taskUuid);
+    }
+    return format("%s/%s/%s.log", taskType, componentUuid, taskUuid);
+  }
+
+  @Override
+  public boolean equals(@Nullable Object o) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/log/LogFileRef.java]: MAJOR The Cyclomatic Complexity of this method "equals" is 11 which is greater than 10 authorized. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by simonbrandhof** [comment]
Feedback applied.

**Pull request closed without merging**